addons/html_editor/static/src/others/embedded_components/plugins/toggle_plugin/toggle_plugin.js

```diff
@@ -0,0 +1,504 @@
+            this.dependencies.dom.insert(fragment);
+        }
+        container.replaceWith(newParagraph);
+        this.dependencies.history.addStep();
+        this.dependencies.selection.setCursorStart(newParagraph);
+        return true;
+    }
+
+    /**
+     * Handles the tab behavior. This means that when we are inside a toggle title and we have a toggle
+     * as previous sibling of the embedded component, the current toggle is indented inside the content of
+     * the previous one.
+     */
+    handleTab() {
+        const selection = this.dependencies.selection.getEditableSelection();
+        const closestToggleTitle = closestElement(selection.anchorNode, titleSelector);
+        if (closestToggleTitle) {
+            // If selection is in a title.
+            const container = closestToggleTitle.closest("[data-embedded]");
+            const previousSibling = container.previousElementSibling;
+            if (previousSibling?.closest("[data-embedded='toggle']") === previousSibling) {
+                // If the previous element of the embedded component is also a toggle, we need to handle
+                // it.
+                const cursors = this.dependencies.selection.preserveSelection();
+                this.dependencies.selection.setCursorEnd(
+                    previousSibling.querySelector("[data-embedded-editable='content']")
+                );
+                previousSibling.firstChild.querySelector("i.fa-caret-right")?.click(); // open the toggle if needed
+                const fragment = this.document.createDocumentFragment();
+                fragment.replaceChildren(container);
+                const canReplace = previousSibling.querySelector(
+                    "[data-embedded-editable='content'] > *:only-child"
+                );
+                // If the 1st block of the previous content is empty we replace it with our toggle.
+                // Else we add it to the content.
+                if (isEmptyBlock(canReplace)) {
+                    canReplace.replaceWith(fragment);
+                } else {
+                    this.dependencies.dom.insert(fragment);
+                }
+                this.dependencies.history.addStep();
+                window.setTimeout(cursors.restore, "animationFrame"); // Used to handle caret displaying issues
+            }
+            return true;
+        }
+    }
+
+    /**
+     * Handles the shift-tab behavior. This means that we need to outdent the toggle from each other.
+     * @returns
+     */
+    handleShiftTab() {
+        const selection = this.dependencies.selection.getEditableSelection();
+        const closestToggleTitle = closestElement(selection.anchorNode, titleSelector);
+        if (closestToggleTitle) {
+            // We are indeed inside a title.
+            const container = closestToggleTitle.closest("[data-embedded]");
+            if (container.parentElement.closest("[data-embedded='toggle']")) {
```

Comment:
I feel like there are some parts that we do in several handlers. For example get the toggle container if the cursor is inside a toggle title. If you confirm this feeling, might be worth to write some helper methods. Would improve readability I think.